### DISCUSSION TITLE
Python - List all users before I can find spesific user?

### DISCUSSION THREAD

**Discussion by GD-Dal** [extremenetworks/ExtremeCloudIQ-APIs#9] [Q&A] [answered]
I'm trying create a python script to list all (PPSK) users to find a specific one. I do this to find out if a specific exists. If it exists -> Update. Else, create new.
But my script lists only 10 users.

Then I see in your script XIQ-PPSKcreateuser.py (which is full of errors, btw) here on Github that you list 10 and 10 users into a greater list.

But if I potentially have 1000 of users, isn't that a bit cumbersome?

If I already know the username (or phone number), is there a way to list only that user?

Thanks.

**Comment by timjsmith24** [comment, marked as answer]
Currently there is no way to get a specific PPSK user id by username or phone number. 
When collecting PPSK Users you can add a limit to the api call to collect 100 users at a time by adding limit=100 to the url. By default it will only return 10. If you have 1000+ ppsk users you will have to page through 100 at at time.
https://api.extremecloudiq.com/endusers?page=1&limit=100
Here's an example of over 1000K PPSK users. So with this information I know I need to collect 19 pages [19 API calls]
{
  "page": 1,
  "count": 100,
  "total_pages": 19,
  "total_count": 1802,
  "data": [
...


As far as that example scripts, I think they were mostly created  to show concepts. But yes, it looks like they haven't been updated for 9 months. These Endpoints were changed a couple months ago so these API calls won't even work.
You could use the loop function from that script tho to collect all PPSK users. Personally, I would create a total_page variable set to 1. then update it after the first call with the number in the response. And I would change the while loop to be 'while page <= total_page:" - The way that script it would stop at 100K users. I can send you an example if you would like.

**Reply by GD-Dal** [reply]
Thank you. But that isn't necessary. I updated my script to use the method you explained.
But something similar to a WHERE option would be very useful in the future versions of the API, though.

idea hereby planted :)
